### PR TITLE
[WIP] Sinker benchmark

### PR DESCRIPTION
**Pull request by gassmoeller**
This is a preliminary implementation of the NSinker benchmark as used in Rudi et al. 2017 (based on May et al 2015). I do not know if I find the time to finish the test during the hackathon, this is mostly to allow @tjhei to take a look.

### CODE REVIEW

**Comment by tjhei** [comment]
I just looked at this PR again and I think we can merge this as is. @gassmoeller what do you think?

I would like to have this in mainline soon, so we can further test the solver...

**Comment by gassmoeller** [comment]
The manual text was outdated and partly wrong (I never found the time to finish it), so I removed it for now from the PR (I will open another one for the text, I still have it in a local branch). The remaining part of this PR should be ready once the tester has finished.

File: benchmarks/nsinker/nsinker.cc [hunk 1]
```diff
@@ -0,0 +1,291 @@
...
+      // Parameters for Sinker example
+      double delta = 200.0;
+      double omega = 0.1;
+      double chi = 1.0;
+
+      // Centers for the sinkers provided by Cedric Thielot (pers. comm. from Dave May)
+      std::vector<Point<dim> > centers(76);
+      if (dim == 3)
+        {
+          centers[1] = Point<3>(2.4257829890e-01, 1.3469574514e-02, 3.8313885004e-01);
```
Comment (tjhei) [inline benchmarks/nsinker/nsinker.cc]: why do you not fill [0] and not use [0]?

File: benchmarks/nsinker/nsinker.cc [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+      const unsigned dim = 3;
+
+      // Parameters for Sinker example
+      double delta = 200.0;
+      double omega = 0.1;
+      double chi = 1.0;
+
+      // Centers for the sinkers provided by Cedric Thielot (pers. comm. from Dave May)
+      std::vector<Point<dim> > centers(76);
+      if (dim == 3)
```
Comment (tjhei) [inline benchmarks/nsinker/nsinker.cc]: this is a noop, right?

**Comment by tjhei** [comment]
Looks good otherwise.

**Comment by tjhei** [comment]
and thank you for looking at it again!

**Comment by gassmoeller** [comment]
> why do you not fill [0] and not use [0]?

Because I got it numbered like this and was lazy. I fixed it now :smile:.

Addressed your comments.

**Pull request merged**